### ISSUE TITLE
Hi ! I want to run the minerl or malmo locally. Is there some solutions for me ?

### ISSUE THREAD

**Issue by yang9004** [minerllabs/minerl#412]
Due to some company network policy, I cann't connect to the interenet all the time.
in other word, can i run the env in offline mode ?

also, every time i start the minerl, it will take almost 1 min to go , comparing to other env, like gym, ithor ,  the minerl is a slow env, can it start more faster ? 

thanks

**Comment by Miffyli** [comment]
There shouldn't be anything in MineRL that connects to the internet, or at least it should still work without internet. You could try running the env when you have internet disabled.

MineRL compiles (and caches some of it) the game every time you launch it and sets up a bunch of mods which is not done in an instant, so long launch time are to be expected. Same goes for `env.reset()`, which creates a new Minecraft world from zero which also takes a short while.

**Comment by yang9004** [comment]
Yes, after I installed MineRL with internet, it can work very well . But when I cut off the internet it just stuck and suggest Malmo error all the time.  I will append some error information after I go back to company.

By the way, I used to use the Malmo python interface and I find it run, start, reset in very short time, because Malmo run the server and agent code in separate program . However, the MineRL seems to take very long time to start (>1 min), especially when I try to update my agent code and test it (the Minerl python code will start a hole new server to run the agent code ). do you have better way to do it ?

**Comment by Miffyli** [comment]
Ah yes, indeed it does not seem to work without internet as it downloads some Forge mods upon compiling the game :/. I do not know of a way around this unfortunately.

MineRL indeed launches both server and client every time you create the environment. This is because MineRL relies on traditional Minecraft mods and partially on Malmo (but looking at the commits I see the authors are moving away from Malmo completely). I do not know of a way to avoid this, but to be frank it is not _that_ long of a wait compared to the speed of the environment in general.

**Comment by yang9004** [comment]
Thank you! If I find any solution I will share it here ...
